packages/components/pie-notification/src/index.ts

```diff
@@ -1,24 +1,247 @@
-import { LitElement, html, unsafeCSS } from 'lit';
-
-import { defineCustomElement } from '@justeattakeaway/pie-webc-core';
+import {
+    unsafeCSS,
+    nothing,
+    LitElement,
+    type TemplateResult,
+    type PropertyValueMap,
+} from 'lit';
+import { type StaticValue, html, unsafeStatic } from 'lit/static-html.js';
+import { defineCustomElement, validPropertyValues } from '@justeattakeaway/pie-webc-core';
+import { property, queryAssignedElements } from 'lit/decorators.js';
+import { type NotificationProps, variants, headingLevels } from './defs';
 import styles from './notification.scss?inline';
-import { NotificationProps } from './defs';
+
+import '@justeattakeaway/pie-icon-button';
+import '@justeattakeaway/pie-icons-webc/IconClose';
+import '@justeattakeaway/pie-icons-webc/IconInfoCircle';
+import '@justeattakeaway/pie-icons-webc/IconAlertCircle';
+import '@justeattakeaway/pie-icons-webc/IconAlertTriangle';
+import '@justeattakeaway/pie-icons-webc/IconCheckCircle';
 
 // Valid values available to consumers
 export * from './defs';
 
 const componentSelector = 'pie-notification';
+const componentClass = 'c-notification';
+
+type InternalVariantType = Exclude<NotificationProps['variant'], undefined>;
+type InternalHeadingLevelType = Exclude<NotificationProps['headingLevel'], undefined>;
 
 /**
  * @tagname pie-notification
  */
 export class PieNotification extends LitElement implements NotificationProps {
-    render () {
-        return html`<h1 data-test-id="pie-notification">Hello world!</h1>`;
-    }
+    @property({ type: Boolean })
+    public isOpen = true;
+
+    @property()
+    @validPropertyValues(componentSelector, variants, 'neutral')
+    public variant: InternalVariantType = 'neutral';
+
+    @property({ type: Boolean })
+    public isCompact = false;
+
+    @property({ type: String })
+    public heading!: string;
+
+    @property()
+    @validPropertyValues(componentSelector, headingLevels, 'h2')
+    public headingLevel: InternalHeadingLevelType = 'h2';
+
+    @property({ type: Boolean })
+    public hideIcon = false;
+
+    @property({ type: Boolean })
+    public hideCloseIcon = false;
+
+    @queryAssignedElements({ slot: 'icon' }) _iconSlot!: Array<HTMLElement>;
+
+    private _hasExternalIcon = false;
+
+    private _hasIconClass = false;
 
     // Renders a `CSSResult` generated from SCSS by Vite
     static styles = unsafeCSS(styles);
+
+    /**
+     * Lifecycle method executed after component renders.
+     */
+    protected firstUpdated (): void {
+        this.updateIconProperties();
+
+        if (this._hasExternalIcon || this._hasIconClass) {
+            this.requestUpdate();
+        }
+    }
+
+    /**
+     * Lifecycle method executed when component is about to update.
+     * It update icon properties if variant has changes.
+     */
+    protected willUpdate (_changedProperties: PropertyValueMap<NotificationProps>): void {
```

Comment:
You can also call `PropertyValues<this>` for a similar effect (It's a slightly different import from Lit). I am unsure which is preferable!